packages/core/src/awsService/cloudWatchLogs/commands/tailLogGroup.ts

```diff
@@ -3,17 +3,142 @@
+import { ToolkitError } from '../../../shared'
 
-export async function tailLogGroup(logData?: { regionName: string; groupName: string }): Promise<void> {
+export async function tailLogGroup(
+    registry: LiveTailSessionRegistry,
+    logData?: { regionName: string; groupName: string }
+): Promise<void> {
     const wizard = new TailLogGroupWizard(logData)
     const wizardResponse = await wizard.run()
     if (!wizardResponse) {
         throw new CancellationError('user')
     }
 
-    //TODO: Remove Log. For testing while we aren't yet consuming the wizardResponse.
-    getLogger().info(JSON.stringify(wizardResponse))
+    const liveTailSessionConfig: LiveTailSessionConfiguration = {
+        logGroupName: wizardResponse.regionLogGroupSubmenuResponse.data,
+        logStreamFilter: wizardResponse.logStreamFilter,
+        logEventFilterPattern: wizardResponse.filterPattern,
+        region: wizardResponse.regionLogGroupSubmenuResponse.region,
+    }
+    const session = new LiveTailSession(liveTailSessionConfig)
+    if (registry.has(session.uri)) {
+        await prepareDocument(session)
+        return
+    }
+    registry.set(session.uri, session)
+
+    const document = await prepareDocument(session)
+    registerTabChangeCallback(session, registry, document)
+    const stream = await session.startLiveTailSession()
+
+    await handleSessionStream(stream, document, session)
+}
+
+export function closeSession(sessionUri: vscode.Uri, registry: LiveTailSessionRegistry) {
+    const session = registry.get(sessionUri)
+    if (session === undefined) {
+        throw new ToolkitError(`No LiveTail session found for URI: ${sessionUri.toString()}`)
+    }
+    session.stopLiveTailSession()
+    registry.delete(sessionUri)
+}
+
+export async function clearDocument(textDocument: vscode.TextDocument) {
+    const edit = new vscode.WorkspaceEdit()
+    const startPosition = new vscode.Position(0, 0)
+    const endPosition = new vscode.Position(textDocument.lineCount, 0)
+    edit.delete(textDocument.uri, new vscode.Range(startPosition, endPosition))
+    await vscode.workspace.applyEdit(edit)
+}
+
+async function prepareDocument(session: LiveTailSession): Promise<vscode.TextDocument> {
+    const textDocument = await vscode.workspace.openTextDocument(session.uri)
+    await clearDocument(textDocument)
+    await vscode.window.showTextDocument(textDocument, { preview: false })
+    await vscode.languages.setTextDocumentLanguage(textDocument, 'log')
+    return textDocument
+}
+
+async function handleSessionStream(
+    stream: AsyncIterable<StartLiveTailResponseStream>,
+    document: vscode.TextDocument,
+    session: LiveTailSession
+) {
+    try {
+        for await (const event of stream) {
+            if (event.sessionUpdate !== undefined) {
+                const formattedLogEvents = event.sessionUpdate.sessionResults!.map<string>((logEvent) =>
```

Comment:
It looks like you're using `!` in `sessionResults!` to assert that sessionResults is never undefined. Is that actually the case? Technically if sessionResults returns undefined one time then it results in an error

Reply:
You're right! I think I had this for some early testing and never when back. I'll add `event.sessionUpdate.sessionResults !== undefined` to the if check.